app/buck2_action_impl/src/context/dynamic_output.rs

```diff
@@ -75,6 +77,9 @@ pub(crate) fn analysis_actions_methods_dynamic_output(methods: &mut MethodsBuild
     fn dynamic_output<'v>(
         this: &'v AnalysisActions<'v>,
         #[starlark(require = named)] dynamic: UnpackListOrTuple<UnpackArtifactOrDeclaredArtifact>,
+        // TODO(ah) consider merging with `dynamic`.
+        #[starlark(require = named, default = UnpackListOrTuple::default())]
+        promises: UnpackListOrTuple<DynamicValue>,
```

Comment:
This is a breaking API change. The lambda that a user passes to `dynamic_output` now needs to take an additional positional argument. A problem is that it produces quite confusing errors on code that assumes the old API, because one of the commonly following keyword arguments with assigned default will be overridden with the additional positional argument. I'm open to suggestions for a different API. Perhaps this could be merged into the `dynamic` argument.